src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -532,4 +548,8 @@ public TruffleNFIPlatform getTruffleNFI() {
     public NativeConfiguration getNativeConfiguration() {
         return nativeConfiguration;
     }
+
+    public Random getRandom() {
+        return random;
```

Comment:
I would expose a minimal interface here so it's clearly only used for seeding.
Something like `getSeedBytes(byte[])` or so.

Reply:
Then we'd couple our `BCrypt` class to `RubyContext`. At the moment it's decoupled via `Random`.

Reply:
BCrypt could also just use `getSeedBytes(int bytes)` or so.

Reply:
> Then we'd couple our `BCrypt` class to `RubyContext`. At the moment it's decoupled via `Random`.
 
Right. I don't think it's a problem.

Reply:
You could have a RandomSeeds class or so if you really wanted to decouple.

Reply:
What does all this achieve? Why does the interface need to be so minimal? I don't want to introduce extra classes just to restrict the methods available for a very trivial class.

Reply:
I think interfaces with just one implementation that you will never implement any other way is a sign you're architecting for the sake of it and need to simplify.

Reply:
Agreed, so that's why I proposed exposing just one method to get a random seed of `n` bytes.
A getter is tight coupling.

Reply:
So you don't mind coupling `BCrypt` to `RubyContext`?

Reply:
No. It's a Ruby library/extension after all.

Reply:
(And we should really just use the C-ext)